builder/vsphere/driver/vm.go

```diff
@@ -339,6 +347,115 @@ func (vm *VirtualMachine) Clone(ctx context.Context, config *CloneConfig) (*Virt
 	return created, nil
 }
 
+func (vm *VirtualMachine) updateVAppConfig(newProps map[string]string) (*types.VmConfigSpec, error) {
+	if len(newProps) == 0 {
+		return nil, nil
+	}
+
+	vProps, _ := vm.Properties()
+	if vProps.Config.VAppConfig == nil {
+		return nil, fmt.Errorf("this VM lacks a vApp configuration and cannot have vApp properties set on it")
+	}
+
+	allProperties := vProps.Config.VAppConfig.GetVmConfigInfo().Property
+
+	var props []types.VAppPropertySpec
+	for _, p := range allProperties {
+		if *p.UserConfigurable == true {
```

Comment:
I think we could reduce the indentation here for readability:

```suggestion
	for _, p := range allProperties {
		userValue, setByUser := newProps[p.Id]
		if !setByUser {
			continue
		}
		if *p.UserConfigurable == false
			return nil, fmt.Errof("%s is not user configurable", p.Id)
		}
		// ...
```